src/index.js

```diff
@@ -1 +1,102 @@
+    if (isError) {
+      messageElement.classList.add('error-message');
+    }
+    messageContainer.appendChild(messageElement);
+
+    // Remove the message element after 3 seconds
+    setTimeout(() => {
+      messageElement.remove();
+    }, 3000);
+  };
+
+  // Function to get scores for the created game
+  const getScores = async () => {
+    try {
+      const response = await fetch(`${baseURL}games/${gameId}/scores`);
+      const data = await response.json();
+      scoresList.innerHTML = ''; // Clear previous scores
+      data.result.forEach((score) => {
+        const listItem = document.createElement('li');
+        listItem.textContent = `${score.user} : ${score.score}`;
+        scoresList.appendChild(listItem);
+      });
+    } catch (error) {
+      displayMessage('Error fetching scores', true);
+    }
+  };
+
+  // Immediately fetch scores when the page is loaded
+  getScores();
+
+  // Function to create a new game
+  const displayScores = async () => {
+    try {
+      const response = await fetch(`${baseURL}games`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ name: 'Leaderboard' }), // Replace with your game name
+      });
+      const data = await response.json();
+      gameId = data.result;
+      getScores(); // Fetch and display scores after creating the game
+    } catch (error) {
+      displayMessage('Error creating game', true);
+    }
+  };
+  // Function to submit a score for the created game
+  const submitScore = async (name, score) => {
+    try {
+      await fetch(`${baseURL}games/${gameId}/scores`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({ user: name, score: parseInt(score, 10) }), // Added radix parameter
+      });
+
+      // Display success message and refresh scores
+      displayMessage('Scores submitted successfully');
+      getScores(); // Refresh scores after submission
+    } catch (error) {
+      displayMessage('Error submitting score', true);
+    }
+  };
+
+  // Event listeners
+  document.querySelector('.refresh').addEventListener('click', () => {
+    window.location.reload();
+  });
+
+  document.querySelector('#form').addEventListener('submit', async (event) => {
+    event.preventDefault();
+    const name = document.querySelector('.name').value;
+    const score = document.querySelector('.Scores').value;
+    if (name && score) {
+      // Use await here to ensure submission completes before refreshing
+      await submitScore(name, score);
+      // Clear input fields after submission
+      document.querySelector('.name').value = '';
+      document.querySelector('.Scores').value = '';
+    } else {
+      displayMessage('Please enter both name and score', true);
+    }
+  });
+
+  // Initialize the game and fetch scores when the page is loaded
+  displayScores();
+  getScores();
+});
```

Comment:
-  [ ] Kindly follow the instructions for creating the project. From the video below, when a score is added, it is immediately added to the page. This renders the refresh button useless. Kindly refactor your app so that the added scores are not added to the page immediately. Only when we click the refresh button should scores be fetched from the API.

- [ ] You seem to be using a different student's API endpoint from a different cohort. When add a score and click the refresh button I no longer see my previously added score, instead the page is filled with unnecessary data from a different API endpoint. Kindly fix this. The app should work such that when I submit the score it is added to the API, and when I click the refresh button the scores I added are correctly fetched. Kindly go through the project guidelines again.

- [ ] When the app is loaded, it fetches scores from an old API endpoint that has been used by previous students from older cohorts. I don't think that's appropriate; it also makes your project UI page overloaded with unnecessary information. Typically you should have your own API endpoint not copy from other people. Kindly go through the leaderboard API documentation in the project guidelines and learn how to retrieve your own information and work with your own, instead of using other people's API.

  [screen-capture (5).webm](https://github.com/richie1988/Leader-Board/assets/36542986/49da3295-acfb-4f9a-828a-331661604d59)